apps/03TicTacToe.js

```diff
@@ -59,53 +72,108 @@ function diagonalWin() {
         playerTurn === board[2][0];
 
 }
+var valid = ['0', '1', '2'];
 
 function ticTacToe(row, column) {
-    board[row][column] = playerTurn;
-    checkForWin();
-    if (checkForWin()) {
-        playAgain();
-    } else {
-        playerTurn = (playerTurn === 'X') ? 'O' : 'X';
-        getPrompt();
+      board[row][column] = playerTurn;
+      checkForWin();
+      if(checkForWin()) {
+          playAgain();
+      } else {
+          playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+          getPrompt();
+      }
+  }
+
+
+
+  // var win = (horizontalWin() || verticalWin() || diagonalWin());
+
+
+
+  var blank = ' ';
+  function isBlank(blank) {
+    return blank;
+  }
+  var blankSpaces = board.length - board.filter(isBlank);
+
+//   function checkForTie()  {
+//   if (!(win)  &&  (blankSpaces > 7)) {
+//   return true;
+// }
+// }
+
+function checkForTie(){
+  for (var i=0;i<board.length;i++){
+    for (var j=0;j<board.length;j++) {
+      if (board[i][j] === ' ') {
+        return false;
+      }
     }
+  }
+  return true;
 }
 
+
+
 function checkForWin() {
-    var win = (horizontalWin() || verticalWin() || diagonalWin());
-    if (win) {
+    var gameOver = false;
+    if (horizontalWin() || verticalWin() || diagonalWin()) {
         console.log(playerTurn + " " + "is the winner!");
         return true;
     }
-
+    else if (checkForTie()) {
+        console.log('Looks like this match is a stalemate.');
```

Comment:
Zac! Awesome work on this app. I tried to break it several ways, and it wouldn't let me. Some suggestions I have would be to look into this 'Looks like this match is a stalemate', i am seeing that message pop up twice when I create a stalemate. But you went above and beyond the criteria for the project, so good work!

![vageta](https://media.giphy.com/media/B9burJ6eRFNJe/giphy-downsized-large.gif)